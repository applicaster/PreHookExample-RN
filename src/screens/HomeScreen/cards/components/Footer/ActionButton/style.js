import { StyleSheet } from 'react-native';

const ICON_SIZE = 30;

export const styles = StyleSheet.create({
  actionButton: {
    flexDirection: 'row',
    height: 30,
    alignItems: 'center',
    marginRight: 7,
  },
  icon: {
    height: ICON_SIZE,
    width: ICON_SIZE,
    marginRight: 1,
  },
  label: {
    fontSize: 12,
  },
});
