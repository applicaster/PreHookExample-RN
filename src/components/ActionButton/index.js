import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { styles } from './style';

class ActionButton extends Component {
  render() {
    const { imageUri, label, onPress } = this.props;
    const { textColor = '#FFFFFF' } = this.context;
    const tintColorStyle = { tintColor: textColor };
    const textColorStyle = { color: `${textColor}99` };

    return (
      <View style={styles.actionButton}>
        <TouchableOpacity onPress={onPress}>
          <Image source={{ uri: imageUri }} style={[styles.icon, tintColorStyle]} />
        </TouchableOpacity>
        <Text style={[styles.label, textColorStyle]} >{label}</Text>
      </View>
    );
  }
}

ActionButton.propTypes = {
  label: PropTypes.number,
  imageUri: PropTypes.string,
  onPress: PropTypes.func,
};

ActionButton.contextTypes = {
  textColor: PropTypes.string,
};

export default ActionButton;
