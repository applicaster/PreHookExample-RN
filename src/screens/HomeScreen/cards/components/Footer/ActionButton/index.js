import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Text,
  TouchableOpacity,
} from 'react-native';
import { Image } from 'react-native-animatable';
import { styles } from './style';

class ActionButton extends Component {
  render() {
    const { imageUri, label, onPress, selected } = this.props;
    const { mainColor = '#FFFFFF', textColor = '#FFFFFF' } = this.context;
    const tintColorStyle = { tintColor: (selected) ? mainColor : textColor };
    const textColorStyle = { color: `${textColor}99` };

    const ANIMATION_TYPE = 'rubberBand';
    const ANIMATION_DURATION = 0;
    const ANIMATION_DELAY = 400;
    return (
      <TouchableOpacity style={styles.actionButton} onPress={onPress}>
        <Image
          animation={ANIMATION_TYPE}
          delay={ANIMATION_DELAY}
          duration={ANIMATION_DURATION}
          source={{ uri: imageUri }}
          style={[styles.icon, tintColorStyle]}
        />
        <Text style={[styles.label, textColorStyle]} >{label}</Text>
      </TouchableOpacity>
    );
  }
}

ActionButton.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  imageUri: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired,
};

ActionButton.contextTypes = {
  mainColor: PropTypes.string,
  textColor: PropTypes.string,
};

export default ActionButton;
