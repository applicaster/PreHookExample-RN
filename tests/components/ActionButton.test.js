import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ActionButton from '../../src/components/ActionButton';

describe('ActionButton', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <ActionButton label={'hi'} source={'some_image_uri'} onPress={() => {}} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when button is selected', () => {
    const html = renderer.create(
      <ActionButton label={'hi'} source={'some_image_uri'} onPress={() => {}} selected />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
