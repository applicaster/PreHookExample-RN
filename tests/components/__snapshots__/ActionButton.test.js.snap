// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "marginRight": 7,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "marginRight": 1,
            "width": 30,
          },
          Object {
            "tintColor": "#FFFFFF",
          },
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  >
    hi
  </Text>
</View>
`;
